### ISSUE TITLE
Updating a non-existing resource causes an internal sever error

### ISSUE THREAD

**Issue by bzar** [ckan/ckan#6925]
**CKAN version**
master

**Describe the bug**
Trying to update a resource that does not exist causes an internal server error instead of a not found error

**Steps to reproduce**
- using `httpie`: `http POST https://demo.ckan.org/api/action/resource_update id=doesnotexist`
- Causes an internal server error

**Expected behavior**
- "Not found" error

**Additional details**
Caused by [this line](https://github.com/ckan/ckan/blob/master/ckan/logic/action/update.py#L76) being placed before [these lines](https://github.com/ckan/ckan/blob/master/ckan/logic/action/update.py#L78-L80).

**Comment by TomeCirun** [comment]
Hey, @bzar, I just tested this via curl, I sent an id that doesn't exist, and it gives me the proper error message: 404 not found.
https://github.com/ckan/ckan/blob/7586d78682c30f205027522214f33ee2bf413055/ckan/logic/action/update.py#L73-L76
if you look at the code you can see in line 73 we have another check

```python 
if resource is None
```

and here are my logs:
![image](https://user-images.githubusercontent.com/72216462/174585256-fd39dc89-87cd-4591-9611-9242e68aa163.png)

![image](https://user-images.githubusercontent.com/72216462/174585391-8d0d13f4-c4ca-4ed4-81b3-71868305818d.png)

**Comment by bzar** [comment]
Yeah, my bad. I tested this on our 2.9.5 instance and demo.ckan.org, saw the same order for those lines in master and assumed it was still an issue.
